### ISSUE TITLE
Go PT Ultra: Registration with reolink servers failed.

### ISSUE THREAD

**Issue by tomaspre** [QuantumEntangledAndy/neolink#167]
**Describe the bug**
When I try to launch Neolink with a cellular camera config, I keep getting _Registration with reolink servers failed_.

**Config**
```
bind = "0.0.0.0"

[[cameras]]
name = "Chata1"
username = "admin"
password = "redacted"
discovery = "cellular"
uid = "redacted
```

**Log**
```
root@neolink-chata:~# export RUST_LOG="neolink=debug"
root@neolink-chata:~# set RUST_LOG "neolink=debug"
root@neolink-chata:~# ./neolink image --config=neolink.toml --file-path=/root/img.jpg
[2023-10-06T08:15:27Z INFO  neolink] Neolink e47a0d5734b0a06b53bd8d78be9ca2f6b8396480 release
[2023-10-06T08:15:27Z DEBUG neolink::common::reactor] Inserting new insance
[2023-10-06T08:15:27Z INFO  neolink::utils] Chata1: Connecting to camera at UID: redacted
[2023-10-06T08:15:27Z DEBUG neolink::common::reactor] New instance created
[2023-10-06T08:15:27Z DEBUG neolink::common::reactor] Got instance from reactor
[2023-10-06T08:15:27Z DEBUG neolink::image] Using the snap command
[2023-10-06T08:15:27Z DEBUG neolink::common::neocam] Chata1: InUse
[2023-10-06T08:15:42Z INFO  neolink_core::bc_protocol] Chata1: Registration with reolink servers failed. Retrying: 1/10
[2023-10-06T08:15:58Z INFO  neolink_core::bc_protocol] Chata1: Registration with reolink servers failed. Retrying: 2/10
[2023-10-06T08:16:14Z INFO  neolink_core::bc_protocol] Chata1: Registration with reolink servers failed. Retrying: 3/10
[2023-10-06T08:16:30Z INFO  neolink_core::bc_protocol] Chata1: Registration with reolink servers failed. Retrying: 4/10
[2023-10-06T08:16:46Z INFO  neolink_core::bc_protocol] Chata1: Registration with reolink servers failed. Retrying: 5/10
[2023-10-06T08:17:02Z INFO  neolink_core::bc_protocol] Chata1: Registration with reolink servers failed. Retrying: 6/10
```

**To Reproduce**
1. (Disable Pihole)
2. Launch Neolink with the above config

**Versions**
NVR software: None, just trying to get an image. The same thing happens when launching with RTSP.
Neolink software: 0.6.2
Reolink camera model and firmware: Go PT Ultra, FW: 3.0.0.2329_23061311 (most recent)

**Details**
- Neolink works fine with wifi cameras. This is my first cellular one. 
- I can capture the traffic with tcpdump if needed. 
- I'm running neolink in an LXC on Proxmox.
- The camera was set up using the Reolink Android app
- The UID and password works fine in the app and the PC client

**Comment by QuantumEntangledAndy** [comment]
Can you capture the connection data with the official client. Will compare it to what we do

**Comment by QuantumEntangledAndy** [comment]
Recently I've seen some cameras not report valid registration details from Reolink. It's possible that the official client has changed the protocol a little bit and I'll need to adjust how we do it

**Comment by tomaspre** [comment]
I captured the data. Can I send you a private message somewhere? I'm a bit too paranoid to post a dump from my network publicly.

**Comment by QuantumEntangledAndy** [comment]
Grab my email from GitHub I suppose.

**Comment by QuantumEntangledAndy** [comment]
You could also install the our wireshark dissector then filter on baichan packets

**Comment by tomaspre** [comment]
![st,small,845x845-pad,1000x1000,f8f8f8](https://github.com/QuantumEntangledAndy/neolink/assets/6633873/6deffb18-cfd5-4e74-8091-6a90ea8576a2)

I solved the issue. It had no connection to neolink whatsoever. My Proxmox host decided not to let one particular LXC use DNS (an identical container with neolink worked while the one with the errors above would not). I'm sorry I wasted your time and thanks so much for the amazing software!!
